src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -0,0 +1,642 @@
+
+    * D. J. Singh and L. Nordström, \"Role of the Linearization Energies,\" in Planewaves, pseudopotentials, and the LAPW method, 2nd ed. New York, NY: Springer, 2006, pp. 49-52.
+    """
+
+    type = Quantity(
+        type=MEnum('lo', 'LO', 'custom'),
+        description="""
+        Type of augmentation contribution. Abbreviations stand for:
+        | name | description | radial product |
+        |------|-------------|----------------|
+        | lo   | 2-parameter local orbital | $A_l u_l (r, E_l) + B_l \dot{u}_l (r, E_l^')$ |
+        | LO   | 3-parameter local orbital | $A_l u_l (r, E_l) + B_l \dot{u}_l (r, E_l^') + C_l \dot{u}_l (r, E_l^{''})$ |
+        | custom | local orbital of a different formula |
+
+        * http://susi.theochem.tuwien.ac.at/lapw/
+        """,
+    )
+
+    boundary_order = Quantity(
+        type=np.int32,
+        shape=['n_terms'],
+        description="""
+        Differential order to which the radial wavefunction is matched at the boundary.
+        """,
+    )
+
+    def get_n_terms(self) -> Optional[int]:
+        """Determine the value of `n_terms` based on the lengths of the representative quantities."""
+        return super().get_n_terms(
+            representative_quantities={
+                'energy_parameter',
+                'energy_parameter_n',
+                'differential_order',
+                'boundary_order',
+            }
+        )
+
+    def bo_terms_to_type(self, bo_terms: Optional[int]) -> Optional[str]:
+        """
+        Set the type of the local orbital based on the boundary order.
+        """  # ? include differential_order
+        if bo_terms is None or len(bo_terms) == 0:
+            return None
+        if sorted(bo_terms) == [0, 1]:
+            return 'lo'
+        elif sorted(bo_terms) == [0, 0, 1]:  # ! double-check
+            return 'LO'
+        else:
+            return 'custom'
+
+    @check_normalized
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+        if np.any(np.isneginf(self.boundary_order)):
+            logger.error(
+                '`APWLOrbital.boundary_order` must be completely non-negative.'
+            )
+
+
+class APWLChannel(BasisSet):
+    """
+    Collection of all (S)(L)APW and local orbital components that contribute
+    to a single $l$-channel. $l$ here stands for the angular momentum parameter
+    in the Laplace spherical harmonics $Y_{l, m}$.
+    """
+
+    name = Quantity(
+        type=np.int32,
+        description="""
+        Angular momentum quantum number of the local orbital.
+        """,
+    )
+
+    n_orbitals = Quantity(
+        type=np.int32,
+        description="""
+        Number of wavefunctions in the l-channel, i.e. $(2l + 1) n_orbitals$.
+        """,
+    )
+
+    orbitals = SubSection(sub_section=APWBaseOrbital.m_def, repeats=True)
+
+    def _determine_apw(self) -> dict[str, int]:
```

Comment:
I am not sure why this is private

Reply:
Because it should be called by `normalize`. The LAPW community uses a variety of abbreviations to refer to the basis set as a whole (i.e. our `BasisSetContainer`). `_determine_apw` just provides this name.